OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -128,4 +128,16 @@ void Frame::extendRealizeTopology(SimTK::State& s) const
 {
     Super::extendRealizeTopology(s);
     groundTransformIndex = getCacheVariableIndex("ground_transform");
+}
+void Frame::extendRealizeVelocity(const SimTK::State& s) const
+{
+  Super::extendRealizeVelocity(s);
+
+  SimTK::Value<SimTK::Transform>::downcast(
+      getSystem().getDefaultSubsystem().
+      updCacheEntry(s, groundTransformIndex)).upd()
+          = calcGroundTransform(s);
```

Comment:
This should not be done in realizeVelocity(). It should be done in realizePosition() because it only involves body positions. It will be repeatedly recalculated unnecessarily if velocities get updated while positions remain fixed (and that does happen).